### PR TITLE
prov/tcp: minor refactoring of code 

### PR DESCRIPTION
**Pull request by nikitaxgusev**

### CODE REVIEW

**Comment by shefty** [comment]
bot:aws:retest

File: prov/tcp/src/tcpx_progress.c [hunk 1]
```diff
@@ -565,11 +562,7 @@ int tcpx_get_rx_entry_op_read_rsp(struct tcpx_ep *tcpx_ep)
 	rx_entry = container_of(entry, struct tcpx_xfer_entry,
 				entry);
 
-	memcpy(&rx_entry->hdr, &tcpx_ep->rx_detect.hdr,
-	       (size_t) tcpx_ep->rx_detect.hdr.base_hdr.payload_off);
-	rx_entry->hdr.base_hdr.op_data = TCPX_OP_READ_RSP;
-	rx_entry->rem_len = (rx_entry->hdr.base_hdr.size -
-			     tcpx_ep->rx_detect.done_len);
+	tcpx_get_rx_entry_op(tcpx_ep, rx_entry, TCPX_OP_READ_RSP);
```
Comment (shefty) [inline prov/tcp/src/tcpx_progress.c]: Of the 4 places where the code is updated, only 2 of them match.  Only 1 place sets mrecv_msg_start.  The other 3 set rem_len, which now isn't being set.  And only 3 set the ep.  I would drop this patch.

The function name is also confusing, as we're not 'getting' anything, only initializing some fields.

File: prov/tcp/src/tcpx.h [hunk 2]
```diff
@@ -192,7 +192,7 @@ struct stage_buf {
 struct tcpx_ep {
 	struct util_ep		util_ep;
 	SOCKET			conn_fd;
-	struct tcpx_rx_detect	rx_detect;
+	struct tcpx_rx_detect	cur_rx_msg;
```
Comment (shefty) [inline prov/tcp/src/tcpx.h]: Please rename the struct as well.

File: prov/tcp/src/tcpx_progress.c (hunk 1 above)
Comment (nikitaxgusev) [reply prov/tcp/src/tcpx_progress.c]: @shefty Sorry for confusing, I dropped that patch.

File: prov/tcp/src/tcpx.h (hunk 2 above)
Comment (nikitaxgusev) [reply prov/tcp/src/tcpx.h]: @shefty Done.

**Pull request merged**